numpy/core/src/multiarray/mapping.c

```diff
@@ -2677,9 +2680,9 @@ array_assign_item_slot_impl(HPyContext *ctx, HPy /* PyArrayObject * */ self, HPy
     else {
         HPy view; // PyArrayObject *
         
-        indices[1].value = PyArray_NDIM(self_struct) - 1;
-        indices[1].type = HAS_ELLIPSIS;
-        if (hpy_get_view_from_index(ctx, self, self_struct, &view, indices, 2, 0) < 0) {
+        hpy_indices[1].value = PyArray_NDIM(self_struct) - 1;
+        hpy_indices[1].type = HAS_ELLIPSIS;
+        if (hpy_get_view_from_index(ctx, self, self_struct, &view, hpy_indices, 2, 0) < 0) {
```

Comment:
I needed to duplicate the `indices` struct, since there is a call to legacy `get_item_pointer` for struct dtypes on line 2672 above

Reply:
Should `indices` be updated here too since both are being maintained?

Reply:
No. Only `hpy_indices` is being maintained. `indices` is used only in `get_item_pointer` in a read-only fashion.